consensus/src/pipeline/execution_schedule_phase.rs

```diff
@@ -78,30 +77,29 @@ impl StatelessPipeline for ExecutionSchedulePhase {
         // Call schedule_compute() for each block here (not in the fut being returned) to
         // make sure they are scheduled in order.
         let mut futs = vec![];
-        for b in &ordered_blocks {
+        for block in &ordered_blocks {
+            let prepare_fut = block.get_prepare_fut();
+            let (input_txns, sig_verified_txns) = prepare_fut.await.unwrap().unwrap();
             let fut = self
                 .execution_proxy
                 .schedule_compute(
-                    b.block(),
-                    b.parent_id(),
-                    b.randomness().cloned(),
-                    lifetime_guard.spawn(()),
+                    block,
+                    (input_txns, sig_verified_txns),
+                    CountedRequest::new((), lifetime_guard.clone()),
                 )
                 .await;
             futs.push(fut)
         }
 
         // In the future being returned, wait for the compute results in order.
-        let fut = tokio::task::spawn(async move {
+        let fut = async move {
```

Comment:
?

Reply:
it makes more sense to wait sequentially than out of order? we spent a long time debugging this with daniel's shadowing code because this goes out of order

Reply:
semantically doesn't really matter, right? it's just confusion in the log?

Reply:
doesn't matter unless we add more things to the future inside that depends on strong ordering